src/middlewared/middlewared/plugins/cluster_linux/ctdb_shared_volume.py

```diff
@@ -0,0 +1,232 @@
+        if mount.error:
+            raise CallError(f'{mount.error}')
+
+        return 'SUCCESS'
+
+    @accepts()
+    @job(lock='ctdb_delete_lock')
+    async def delete(self, job):
+
+        """
+        Delete and unmount the shared volume used by ctdb daemon.
+        """
+
+        # nothing to delete if it doesn't exist
+        exists, started = await self.shared_volume_exists_and_started()
+        if not exists:
+            return
+
+        # umount it first
+        umount = await self.middleware.call('ctdb.shared.volume.umount')
+        await umount.wait()
+        if umount.error:
+            raise CallError(f'{umount.error}')
+
+        if started:
+            # stop the volume
+            force = {'force': True}
+            stop = await self.middleware.call('gluster.volume.stop', self.CTDB_VOL_NAME, force)
+            await stop.wait()
+            if stop.error:
+                raise CallError(f'{stop.error}')
+
+        # now delete it
+        data = {'name': self.CTDB_VOL_NAME}
+        delete = await self.middleware.call('gluster.volume.delete', data)
+        await delete.wait()
+        if delete.error:
+            raise CallError(f'{delete.error}')
+
+        return 'SUCCESS'
+
+    @accepts()
+    @job(lock=JOB_LOCK)
+    async def mount(self, job):
+
+        """
+        Mount the ctdb shared volume locally.
+        """
+
+        # We call this method on startup so if we were
+        # to add a call to the `create` method then all
+        # nodes in the cluster would try to create the
+        # shared volume which is unnecessary and will
+        # cause lots of errors since only 1 node in the
+        # cluster needs to create the volume and it gets
+        # propagated to the peers that were specified
+        # during the creation of the shared volume.
+        #
+        # In summary....dont try to be fancy and call the
+        # create method in here or it's going to cause
+        # headaches.
+        exists, started = await self.shared_volume_exists_and_started()
+        if not exists or not started:
+            raise CallError(
+                'The ctdb shared volume does not exist or '
+                'is not started.'
+            )
+
+        path = pathlib.Path(CTDBConfig.CTDB_LOCAL_MOUNT.value)
+        try:
+            # make sure the dirs are there
+            path.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            raise CallError(f'{e}')
+
+        # try to mount it
+        if not path.is_mount():
+            try:
+                command = [
+                    'mount', '-t', 'glusterfs',
+                    'localhost:/' + path.name, path.as_posix(),
+                ]
+                cp = await run(command)
+                if cp.returncode:
+                    raise CallError(f'{cp.stderr.decode().strip()}')
+            except subprocess.CalledProcessError as e:
+                raise CallError(f'{e.stderr.decode().strip()}')
```

Comment:
It will be re-raised as `CallError` automatically in `adapt_exception`, no need to handle it every time